sylvia/tests/remote.rs

```diff
@@ -37,8 +37,10 @@ impl SomeContract {
 // Making sure `Remote` can be stored in `#[cw_serde]` types
 #[cw_serde]
 #[allow(dead_code)]
-struct CustomStorage<Contract> {
-    remote: sylvia::types::Remote<'static, Contract>,
+struct CustomStorage<'a, Contract> {
+    remote: sylvia::types::Remote<'a, Contract>,
+    interface_remote:
+        sylvia::types::Remote<'static, &'a dyn some_interface::SomeInterface<Error = StdError>>,
```

Comment:
Can we use `&'static dyn` here instead of adding a new lifetime to the structure? Let's consider removing the reference in the implementation of Querier for dyn Trait as well.

Reply:
Out of scope for this PR. Created #372 to track it.

Reply:
This is the only thing that needs to be done:
```diff
diff --git a/examples/interfaces/custom-and-generic/src/lib.rs b/examples/interfaces/custom-and-generic/src/lib.rs
index e1f94df..f860b30 100644
--- a/examples/interfaces/custom-and-generic/src/lib.rs
+++ b/examples/interfaces/custom-and-generic/src/lib.rs
@@ -109,7 +109,7 @@ mod tests {
 
         let querier = sylvia::types::BoundQuerier::<
             _,
-            &dyn super::CustomAndGeneric<
+            dyn super::CustomAndGeneric<
                 RetT = SvCustomMsg,
                 Exec1T = SvCustomMsg,
                 Exec2T = SvCustomMsg,
diff --git a/examples/interfaces/generic/src/lib.rs b/examples/interfaces/generic/src/lib.rs
index 782818f..0624268 100644
--- a/examples/interfaces/generic/src/lib.rs
+++ b/examples/interfaces/generic/src/lib.rs
@@ -107,7 +107,7 @@ mod tests {
 
         let querier = sylvia::types::BoundQuerier::<
             Empty,
-            &dyn super::Generic<
+            dyn super::Generic<
                 Exec1T = SvCustomMsg,
                 Exec2T = SvCustomMsg,
                 Exec3T = SvCustomMsg,
diff --git a/sylvia-derive/src/querier.rs b/sylvia-derive/src/querier.rs
index 3e1f61d..788e949 100644
--- a/sylvia-derive/src/querier.rs
+++ b/sylvia-derive/src/querier.rs
@@ -72,7 +72,7 @@ where
                 #(#methods_declaration)*
             }
 
-            impl <'a, C: #sylvia ::cw_std::CustomQuery, #(#all_generics,)*> Querier for #sylvia ::types::BoundQuerier<'a, C, &dyn #interface_name <#( #all_generics = #all_generics,)*> > #where_clause {
+            impl <'a, C: #sylvia ::cw_std::CustomQuery, #(#all_generics,)*> Querier for #sylvia ::types::BoundQuerier<'a, C, dyn #interface_name <#( #all_generics = #all_generics,)*> > #where_clause {
                 #(type #generics = #generics;)*
 
                 #(#methods_trait_impl)*
diff --git a/sylvia/src/types.rs b/sylvia/src/types.rs
index 4ff5e8a..96593a2 100644
--- a/sylvia/src/types.rs
+++ b/sylvia/src/types.rs
@@ -4,13 +4,13 @@ use schemars::JsonSchema;
 use serde::de::DeserializeOwned;
 use serde::{Deserialize, Serialize};
 
-pub struct BoundQuerier<'a, C: cosmwasm_std::CustomQuery, Contract> {
+pub struct BoundQuerier<'a, C: cosmwasm_std::CustomQuery, Contract: ?Sized> {
     contract: &'a cosmwasm_std::Addr,
     querier: &'a cosmwasm_std::QuerierWrapper<'a, C>,
     _phantom: std::marker::PhantomData<Contract>,
 }
 
-impl<'a, C: cosmwasm_std::CustomQuery, Contract> BoundQuerier<'a, C, Contract> {
+impl<'a, C: cosmwasm_std::CustomQuery, Contract: ?Sized> BoundQuerier<'a, C, Contract> {
     pub fn querier(&self) -> &'a cosmwasm_std::QuerierWrapper<'a, C> {
         self.querier
     }

```

Reply:
done

Reply:
Nice! Let's update the example code:
```rust
struct CustomStorage<Contract> {
    remote: sylvia::types::Remote<'static, Contract>,
    interface_remote:
        sylvia::types::Remote<'static, dyn some_interface::SomeInterface<Error = StdError>>,
```